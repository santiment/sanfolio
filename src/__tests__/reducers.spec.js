/* eslint-env jasmine */
import {
  intentForm,
  settings,
  portfolios
} from '../rootReducers'

describe('reducers ', () => {
  it('set amount for the first income money', () => {
    const nextState = intentForm({},
      {
        type: 'INPUT_AMOUNT_MONEY',
        output: 49
      })
    expect(nextState).toMatchSnapshot()
  })

  it('change settings', () => {
    const nextState = settings({realtime: true},
      {
        type: 'CHANGE_SETTINGS',
        realtime: false
      })
    expect(nextState).toMatchSnapshot()
  })

  describe('Portfolio Reducer ', () => {
    it('should create new portfolio and select created portfolio', () => {
      const oldState = {
        selected: 0,
        items: [
          {
            name: 'Portfolio 1'
          },
          {
            name: 'Portfolio 2'
          }
        ]
      }
      const nextState = portfolios(oldState,
        {
          type: 'CREATE_NEW_PORTFOLIO',
          name: 'Portfolio 3',
          items: [],
          money: 23440
        })
      expect(nextState.selected).toBe(2)
      expect(nextState.items.length).toBe(3)
      expect(nextState).toMatchSnapshot()
    })

    it('should select the portfolio', () => {
      const oldState = {
        selected: 0,
        items: [
          {
            name: 'Portfolio 1'
          },
          {
            name: 'Portfolio 2'
          }
        ]
      }
      const nextState = portfolios(oldState,
        {
          type: 'SELECT_PORTFOLIO',
          selected: 1
        })
      expect(nextState).toMatchSnapshot()
    })

    it('should remove selected portfolio', () => {
      const oldState = {
        selected: 0,
        items: [
          {
            name: 'Portfolio 1'
          },
          {
            name: 'Portfolio 2'
          }
        ]
      }
      const nextState = portfolios(oldState,
        {
          type: 'REMOVE_SELECTED_PORTFOLIO'
        })
      expect(nextState.items.length).toBe(1)
      expect(nextState).toMatchSnapshot()
    })

    it('should update selected portfolio', () => {
      const oldState = {
        selected: 0,
        items: [
          {
            name: 'Portfolio 1',
            items: [],
            money: 10
          },
          {
            name: 'Portfolio 2'
          }
        ]
      }
      const nextState = portfolios(oldState,
        {
          type: 'UPDATE_SELECTED_PORTFOLIO',
          name: 'Awesome Portfolio',
          items: [],
          money: 200
        })
      expect(nextState.items.length).toBe(2)
      expect(nextState).toMatchSnapshot()
    })

    it('should not select non existing Portfolio', () => {
      const oldState = {
        selected: 1,
        items: [
          {
            name: 'Portfolio 1'
          },
          {
            name: 'Portfolio 2'
          }
        ]
      }
      const nextState = portfolios(oldState,
        {
          type: 'SELECT_PORTFOLIO',
          selected: 3
        })
      expect(nextState).toMatchSnapshot()
    })
  })
})
