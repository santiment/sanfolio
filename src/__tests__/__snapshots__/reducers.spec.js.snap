// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers  Portfolio Reducer  should create new portfolio and select created portfolio 1`] = `
Object {
  "items": Array [
    Object {
      "name": "Portfolio 1",
    },
    Object {
      "name": "Portfolio 2",
    },
    Object {
      "items": Array [],
      "money": 23440,
      "name": "Portfolio 3",
    },
  ],
  "selected": 2,
}
`;

exports[`reducers  Portfolio Reducer  should not select non existing Portfolio 1`] = `
Object {
  "items": Array [
    Object {
      "name": "Portfolio 1",
    },
    Object {
      "name": "Portfolio 2",
    },
  ],
  "selected": 1,
}
`;

exports[`reducers  Portfolio Reducer  should remove selected portfolio 1`] = `
Object {
  "items": Array [
    Object {
      "name": "Portfolio 2",
    },
  ],
  "selected": 0,
}
`;

exports[`reducers  Portfolio Reducer  should select the portfolio 1`] = `
Object {
  "items": Array [
    Object {
      "name": "Portfolio 1",
    },
    Object {
      "name": "Portfolio 2",
    },
  ],
  "selected": 1,
}
`;

exports[`reducers  Portfolio Reducer  should update selected portfolio 1`] = `
Object {
  "items": Array [
    Object {
      "items": Array [],
      "money": 200,
      "name": "Awesome Portfolio",
    },
    Object {
      "name": "Portfolio 2",
    },
  ],
  "selected": 0,
}
`;

exports[`reducers  change settings 1`] = `
Object {
  "realtime": false,
}
`;

exports[`reducers  set amount for the first income money 1`] = `
Object {
  "money": undefined,
}
`;
